src/_utils.ts

```diff
@@ -56,6 +56,23 @@ export function parseGitURI(input: string): GitInfo {
   };
 }
 
+const tarInputRegex =
+  /^(?<url>https?:\/\/[\w./-]+)(#(name=)?)?(?<name>[\w.-]+)?((#(version|v)=)(?<version>[\w.-]+))?/;
+
+export function parseTarURI(input: string): TarInfo {
+  const m = input.match(tarInputRegex)?.groups;
+
+  if (!m?.url) {
+    throw new Error(`Failed to parse tar url: ${input}`);
+  }
+
+  return <TarInfo>{
+    url: m.url,
+    name: m.name || "",
+    version: m.version || "",
```

Comment:
I think that `version` is not used anywhere if both `name` and `version` provided. I would just keep only `name`